tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/ContextPropagation.cs

```diff
@@ -34,9 +35,35 @@ private static void Inject<TMessageRequest>(SpanContext context, IDictionary mes
             messageAttributes[SnsKey] = CachedMessageHeadersHelper<TMessageRequest>.CreateMessageAttributeValue(stream);
         }
 
-        public static void InjectHeadersIntoMessage<TMessageRequest>(IContainsMessageAttributes carrier, SpanContext spanContext)
+        public static void InjectHeadersIntoBatch<TBatchRequest>(IPublishBatchRequest request, SpanContext context)
         {
-            // add distributed tracing headers to the message
+            // Skip adding Trace Context if entries don't exist or empty.
+            if (request.PublishBatchRequestEntries is not { Count: > 0 })
+            {
+                return;
+            }
+
+            for (var i = 0; i < request.PublishBatchRequestEntries.Count; i++)
+            {
+                var entry = request.PublishBatchRequestEntries[i].DuckCast<IContainsMessageAttributes>();
```

Comment:
It's unlikely, but maybe we should guard against `request.PublishBatchRequestEntries[i]` being null?
```suggestion
                var entry = request.PublishBatchRequestEntries[i]?.DuckCast<IContainsMessageAttributes>();
```

Reply:
Will add it!